ospfd/ospf_vty.c

```diff
@@ -6330,18 +6785,19 @@ static int show_ip_ospf_database_common(struct vty *vty, struct ospf *ospf,
 		return CMD_WARNING;
 
 	/* `show ip ospf database LSA'. */
-	if (argc == arg_base + 5)
-		show_lsa_detail(vty, ospf, type, NULL, NULL);
+	if ((argc == arg_base + 5) || (uj && (argc == arg_base + 6)))
+		show_lsa_detail(vty, ospf, type, NULL, NULL, json);
 	else if (argc >= arg_base + 6) {
 		ret = inet_aton(argv[arg_base + 5]->arg, &id);
 		if (!ret)
 			return CMD_WARNING;
 
 		/* `show ip ospf database LSA ID'. */
-		if (argc == arg_base + 6)
-			show_lsa_detail(vty, ospf, type, &id, NULL);
+		if ((argc == arg_base + 6) || (uj && (argc == arg_base + 7)))
+			show_lsa_detail(vty, ospf, type, &id, NULL, json);
```

Comment:
Change
show_lsa_detail(vty, ospf, type, NULL, NULL, json);
to
show_lsa_detail(vty, ospf, type, NULL, NULL, json_vrf);

Reply:
Done